test/test.js

```diff
@@ -19,4 +20,130 @@ describe('Grammar', function() {
       assert.equal(output, false);
     });
   });
+
+
+  describe("#mandatory_markers",function(){
```

Comment:
Let's come-up with a test-plan so that we some idea about our coverage and can make claims.